lib/mindee/parsing/custom/list_field.rb

```diff
@@ -48,16 +49,12 @@ class ListField
         def initialize(prediction, page_id, reconstructed: false)
           @values = []
           @confidence = prediction['confidence']
-          @page_id = if page_id.nil?
-                       prediction.include?('page_id') ? prediction['page_id'] : nil
-                     else
-                       page_id
-                     end
+          page_id = prediction['page_id'] if page_id.nil? && prediction.include?('page_id')
           @reconstructed = reconstructed
 
           prediction['values'].each do |field|
-            @page_id = field['page_id'] if field.include?('page_id')
-            @values.push(ListFieldItem.new(field))
+            page_id = field['page_id'] if field.include?('page_id')
+            @values.push(ListFieldItem.new(field, page_id: page_id))
```

Comment:
no need to handle `page_id` at this level, as ListFieldValue should grab it from the prediciton